### PR TITLE
TF2Forecaster support `tf.data` dataset

### PR DESCRIPTION
**Pull request by liangs6212**

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py [hunk 1]
```diff
@@ -93,8 +99,8 @@ def evaluate(self, data, batch_size=32, multioutput="raw_values"):
                 non-distribtued version.
         """
         if not self.fitted:
-            raise RuntimeError("You must call fit or restore first before calling predict!")
-        yhat = self.internal.predict(data[0], batch_size=batch_size)
+            raise RuntimeError("You must call fit or restore first before calling evaluate!")
+        yhat = self.internal.predict(data[0].batch(batch_size))
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py:103]: change back to `batch_size=batch_size`

File: python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py [hunk 2]
```diff
@@ -63,6 +73,14 @@ def test_lstm_forecaster_fit_predict_evaluate(self):
                                   multioutput="raw_values")
         assert mse[0].shape == test_data[1].shape[1:]
 
+    def test_lstm_forecaster_fit_tf_data(self):
+        train_data, test_data = create_data()
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/forecaster/tf/test_lstm_keras_forecaster.py:77]: `tf_data=True`

File: python/chronos/test/bigdl/chronos/data/test_tsdataset.py [hunk 3]
```diff
@@ -493,6 +493,16 @@ def test_tsdata_multi_unscale_numpy_torch_load(self):
 
         tsdata._check_basic_invariants()
 
+    def test_tsdata_to_tf_dataset(self):
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/data/test_tsdataset.py:496]: add testskip here and only test if for tf>2.0 (only for this method)

File: python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py [hunk 4]
```diff
@@ -39,14 +39,20 @@ def fit(self, data, epochs=1, batch_size=32):
                | should be the same as past_seq_len and input_feature_num.
                | y's shape is (num_samples, horizon, target_dim), where horizon and target_dim
                | should be the same as future_seq_len and output_feature_num.
+               |
+               | 2. a tf.data.Dataset:
+               | Should return a tuple of (x, y),
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/tf/base_forecaster.py:44]: A TFDataset instance which contains x and y with same shape as the tuple.

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Comment by liangs6212** [comment]
http://10.112.231.51:18889/job/BigDL-PRVN-chronos-Python-Spark-3.1-py37-Horovod/350/
http://10.112.231.51:18889/job/BigDL-Chronos-PR-Validation/305/

**Pull request merged**